Fixes issue #4110 Inconvenient behavior of Ctrl+Right and Ctrl+Left

mkrueger opened the pull request:
In theory we could switch to all vs.net editor operations however
we've many mac behavior commands built in so we would break some mac
specific behavior. But it's ok for ctrl+left/right there are no mac
specifics here.

Therzok commented on main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/MonoTextEditor.ITextView.cs:
```diff
@@ -349,8 +353,8 @@ public void DisplayTextLineContainingBufferPosition (SnapshotPoint bufferPositio
-		{
-			throw new NotImplementedException ();
+		{
+			return new SnapshotSpan (point, point);
```
Shouldn't we be looking for a word start/end at a given point?

mkrueger commented:
@Therzok : It's not what this method does. For the common case it's just returning the extent of 1 character. However I added the support of multibyte characters there and implemented the GetTextElementSpan for the line collection and line.

One thing it's currently missing is the support of non printable characters but that's not in caret move actions either. But multibyte should now work.
